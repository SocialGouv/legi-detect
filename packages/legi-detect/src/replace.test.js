const replace = require("./replace");

const SAMPLE_TEXT = `Il était une fois L'Ordonnance n° 2017-95233 du 22 septembre 2017 qui proclama :

1° L'article L. 2232-5 du code du travail est complété par un deuxième alinéa ainsi rédigé :
« Sauf disposition contraire, les termes “ convention de branche ” désignent la convention collective et les accords de branche, les accords professionnels et les accords interbranches. » ;

article L123-11-3 du code du commerce

article 1045 du Code général des impôts

article L1131-1 du code de la défense

à l'article Article L311-13 du Code de l'entrée et du séjour des étrangers et du droit d'asile
2° L'article L. 2232-5-1 est ainsi modifié :
a) Les premier et deuxième alinéas de l'article L. 2232-5-1 sont remplacés par les dispositions suivantes :
« La branche a pour missions :
« 1° De définir les conditions d'emploi et de travail des salariés ainsi que les garanties qui leur sont applicables dans les matières mentionnées aux articles L. 2253-1 et D1143-4 dans les conditions prévues par lesdits articles. » ;
b) Le 2° de l'article L. 2232-5-1 est supprimé ;
c) Le 3° devient le 2° ;

les fameux articles L311-13 et  L311-18 du Code de l'entrée et du séjour des étrangers et du droit d'asile.

3° L'article R131-33 du code pénal est complété par un deuxième alinéa ainsi rédigé :
« Sauf disposition contraire, les termes “ convention d'entreprise ” désignent toute convention ou accord conclu soit au niveau de l'entreprise, soit au niveau de l'établissement. » ;

4° Les articles L. 2253-1 à L. 2253-3 sont remplacés par les dispositions suivantes :

4° Les articles R131-24 à R. 131-27 du code pénal sont remplacés par les dispositions suivantes :

4° L'article L. 123-1 du code de la route est remplacé par l'article R131-2 du code pénal et ainsi de suite`;

it("Long Text should produce correct HTML", () => {
  expect(replace(SAMPLE_TEXT)).toMatchSnapshot();
});
