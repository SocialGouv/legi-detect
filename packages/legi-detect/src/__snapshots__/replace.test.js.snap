// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Long Text should produce correct HTML 1`] = `"Il était une fois L'Ordonnance n° 2017-95233 du 22 septembre 2017 qui proclama :<br><br>abbréviation  <span class=\\"highlight\\" title=\\"Article L6361-5\\">art.  L6361-5</span><br><br>et ( <span class=\\"highlight\\" title=\\"Article L6361-5\\">Article L6361-5</span>)<br><br>Sous-section 1 : Contrôle des dépenses et activités de formation. ( <span class=\\"highlight\\" title=\\"Articles L6361-1 L6361-3 6361-3-3\\">Articles L6361-1 à L6361-3</span>)<br><br>Section 2 : Agents de contrôle. ( <span class=\\"highlight\\" title=\\"Article L6361-5\\">Article L6361-5</span>)<br><br>Section 3 : Dispositions d'application. ( <span class=\\"highlight\\" title=\\"Article L6361-6\\">Article L6361-6</span>)<br><br>Section 1 : Accès aux documents et justifications à apporter. ( <span class=\\"highlight\\" title=\\"Articles L6362-1 L6362-7-3 6362-7-3-3\\">Articles L6362-1 à L6362-7-3</span>)<br><br>Section 2 : Procédure.  <span class=\\"highlight\\" title=\\"Articles L6362-8 L6362-11 6362-11-11\\">Articles L6362-8 à L6362-11</span><br><br>Section 3 : Sanctions. ( <span class=\\"highlight\\" title=\\"Article L6362-12\\">Article L6362-12</span>)<br><br>Section 4 : Dispositions d'application. ( <span class=\\"highlight\\" title=\\"Article L6362-13\\">Article L6362-13</span>)<br><br>Chapitre III : Constatation des infractions et dispositions pénales<br><br>1° L' <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000035627459&cidTexte=LEGITEXT000006072050\\" class=\\"highlight\\" title=\\"article L. 2232-5 du code du travail\\">Article L2232-5 du Code du travail</a> est complété par un deuxième alinéa ainsi rédigé :<br>« Sauf disposition contraire, les termes “ convention de branche ” désignent la convention collective et les accords de branche, les accords professionnels et les accords interbranches. » ;<br><br> <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000020196876&cidTexte=LEGITEXT000005634379\\" class=\\"highlight\\" title=\\"article L123-11-3 du code du commerce\\">Article L123-11-3 du Code de commerce</a><br><br> <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000029738408&cidTexte=LEGITEXT000006069577\\" class=\\"highlight\\" title=\\"article 1045 du Code général des impôts\\">Article 1045 du Code général des impôts</a><br><br> <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000020932642&cidTexte=LEGITEXT000006071307\\" class=\\"highlight\\" title=\\"article L1131-1 du code de la défense\\">Article L1131-1 du Code de la défense</a><br><br>à l'article  <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000037986972&cidTexte=LEGITEXT000006070158\\" class=\\"highlight\\" title=\\"Article L311-13 du Code de l'entrée et du séjour des étrangers et du droit\\">Article L311-13 du Code de l'entrée et du séjour des étrangers et du droit d'asile</a> d'asile<br>2° L' <span class=\\"highlight\\" title=\\"Article L2232-5-1\\">article L. 2232-5-1</span> est ainsi modifié :<br>a) Les premier et deuxième alinéas de l' <span class=\\"highlight\\" title=\\"Article L2232-5-1\\">article L. 2232-5-1</span> sont remplacés par les dispositions suivantes :<br>« La branche a pour missions :<br>« 1° De définir les conditions d'emploi et de travail des salariés ainsi que les garanties qui leur sont applicables dans les matières mentionnées aux  <span class=\\"highlight\\" title=\\"Articles L2253-1 D1143-4 1143-4-4\\">articles L. 2253-1 et D1143-4</span> dans les conditions prévues par lesdits articles. » ;<br>b) Le 2° de l' <span class=\\"highlight\\" title=\\"Article L2232-5-1\\">article L. 2232-5-1</span> est supprimé ;<br>c) Le 3° devient le 2° ;<br><br>les fameux  <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCode.do?idArticle=LEGIARTI000037986972&cidTexte=LEGITEXT000006070158\\" class=\\"highlight\\" title=\\"articles L311-13 et  L311-18 du Code de l'entrée et du séjour des étrangers et du droit\\">Articles L311-13 L311-18 311-18-18 du Code de l'entrée et du séjour des étrangers et du droit d'asile</a> d'asile.<br><br>3° L' <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000006419397&cidTexte=LEGITEXT000006070719\\" class=\\"highlight\\" title=\\"article R131-33 du code pénal\\">Article R131-33 du Code pénal</a> est complété par un deuxième alinéa ainsi rédigé :<br>« Sauf disposition contraire, les termes “ convention d'entreprise ” désignent toute convention ou accord conclu soit au niveau de l'entreprise, soit au niveau de l'établissement. » ;<br><br>4° Les  <span class=\\"highlight\\" title=\\"Articles L2253-1 L. 2253-3 2253-3-3\\">articles L. 2253-1 à L. 2253-3</span> sont remplacés par les dispositions suivantes :<br><br>4° Les  <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCode.do?idArticle=LEGIARTI000006419388&cidTexte=LEGITEXT000006070719\\" class=\\"highlight\\" title=\\"articles R131-24 à R. 131-27 du code pénal\\">Articles R131-24 R. 131-27 131-27-27 du Code pénal</a> sont remplacés par les dispositions suivantes :<br><br>4° L' <span class=\\"highlight\\" title=\\"Article L123-1 du Code de la route\\">article L. 123-1 du code de la route</span> est remplacé par l' <a target=\\"_blank\\" href=\\"https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000006419340&cidTexte=LEGITEXT000006070719\\" class=\\"highlight\\" title=\\"article R131-2 du code pénal\\">Article R131-2 du Code pénal</a> et ainsi de suite"`;
